app/validators/routing/new_when_rule_validator.py

```diff
@@ -0,0 +1,196 @@
+from app.validators.routing.types import (
+    python_type_to_json_type,
+    resolve_value_source_json_type,
+    TYPE_DATE,
+    TYPE_ARRAY,
+    TYPE_NUMBER,
+    TYPE_STRING,
+    TYPE_NULL,
+    TYPE_BOOLEAN,
+)
+from app.validators.validator import Validator
+
+OPERATOR_NOT = "not"
+OPERATOR_AND = "and"
+OPERATOR_OR = "or"
+OPERATOR_EQUAL = "=="
+OPERATOR_NOT_EQUAL = "!="
+OPERATOR_GREATER_THAN = ">"
+OPERATOR_LESS_THAN = "<"
+OPERATOR_GREATER_THAN_OR_EQUAL = ">="
+OPERATOR_LESS_THAN_OR_EQUAL = "<="
+OPERATOR_IN = "in"
+OPERATOR_ALL_IN = "all-in"
+OPERATOR_ANY_IN = "any-in"
+OPERATOR_DATE = "date"
+
+BOOLEAN_OPERATORS = [OPERATOR_NOT, OPERATOR_AND, OPERATOR_OR]
```

Comment:
Boolean always makes me think true false, is logic operators not more appropriate?

Reply:
Happy to go with `LOGIC_OPERATORS`.

Reply:
I have changed this in https://github.com/ONSdigital/eq-questionnaire-validator/pull/97/commits/e7d9c17c0627a151002290fcccd6bee2fde90398. As part of this I have also renamed `boolean` to `logic` in the schemas.